tests/node-e2e/outgoing/test_outgoing_traffic_fails_for_bonkers_hostname.mjs

```diff
@@ -0,0 +1,38 @@
+import https from "node:https";
+
+console.log(">> test_outgoing_traffic_fails_for_bonkers_hostname");
+
+const options = {
+  hostname: "www.bonkers1234.com.net",
+  port: 443,
+  path: "/",
+  method: "GET",
+};
+
+const request = https.request(options, (response) => {
+  console.log(`statusCode: ${response.statusCode}`);
+
+  response.on("data", (data) => {
+    process.stdout.write(data);
+  });
+
+  response.on("error", (fail) => {
+    console.error(`>> response failed with ${fail}`);
+
+    process.exit(-1);
+  });
+
+  if (response.statusCode === 200) {
+    process.exit();
+  } else {
+    process.exit(-1);
+  }
+});
+
+request.on("error", (fail) => {
+  console.error(`>> request failed with ${fail}`);
+
+  process.exit(-1);
+});
+
+request.end();
```

Comment:
aren't the status codes wrong? like it exits successfully on success instead on http failure?